### DISCUSSION TITLE
Linewise animation

### DISCUSSION THREAD

**Discussion by bensapirstein** [coldtype/coldtype#154] [Q&A]
I want to animate a text with three lines, each transforming at different times—similar to Glyphwise but applied to rows. What would be a straightforward implementation for achieving this effect?

**Comment by stenson** [comment]
Interesting question! I don't think something as intense as `Glyphwise` is required if you're only looking to change type-settings per line, so you could just do something a little more straightforward, i.e. iterate over lines in a text block. That said, the `Glyphwise` callback function argument does include an `.l` (which is the line-index), so you could use `Glyphwise` in order to just target each line. Here are two strategies to get to the same result:

```python
from coldtype import *

@animation((1080, 540), tl=60)
def scratch1(f):
    return (Glyphwise("COLD\nTYPE", lambda x:
        Style(Font.ColdObvi(), 200
            , wdth=f.adj(x.l*7).e("eeio")))
        .lead(10)
        .xalign(f.a.r)
        .align(f.a.r))

@animation((1080, 540), tl=60)
def scratch2(f):
    txt = "COLD\nTYPE"
    return (P().enumerate(txt.splitlines(), lambda x:
        StSt(x.el, Font.ColdObvi(), 200
            , wdth=f.adj(x.i*7).e("eeio")))
        .stack(10)
        .xalign(f.a.r)
        .align(f.a.r))

```
